code/modules/shuttle/emergency.dm

```diff
@@ -61,36 +63,64 @@
 		if(!(ACCESS_HEADS in W:access)) //doesn't have this access
 			to_chat(user, "The access level of [W:registered_name]\'s card is not high enough. ")
 			return 0
-
-		var/choice = tgui_alert(user, "Would you like to (un)authorize a shortened launch time? [auth_need - length(authorized)] authorization\s are still needed. Use abort to cancel all authorizations.", "Shuttle Launch", list("Authorize", "Repeal", "Abort"))
-		if(SSshuttle.emergency.mode != SHUTTLE_DOCKED || user.get_active_hand() != W)
-			return 0
-
-		var/seconds = SSshuttle.emergency.timeLeft()
-		if(seconds <= 10)
-			return 0
-
+		if(!SSshuttle.emergency.aihacked)
+			var/choice = tgui_alert(user, "Would you like to (un)authorize a shortened launch time? [auth_need - length(authorized)] authorization\s are still needed. Use abort to cancel all authorizations.", "Shuttle Launch", list("Authorize", "Repeal", "Abort"))
+			if(SSshuttle.emergency.mode != SHUTTLE_DOCKED || user.get_active_hand() != W)
+				return 0
+
+			var/seconds = SSshuttle.emergency.timeLeft()
+			if(seconds <= 10)
+				return 0
+
+			switch(choice)
+				if("Authorize")
+					if(!authorized.Find(W:registered_name))
+						authorized += W:registered_name
+						if(auth_need - length(authorized) > 0)
+							message_admins("[key_name_admin(user)] has authorized early shuttle launch.")
+							log_game("[key_name(user)] has authorized early shuttle launch in ([x], [y], [z]).")
+							GLOB.minor_announcement.Announce("[auth_need - length(authorized)] more authorization(s) needed until shuttle is launched early")
+						else
+							message_admins("[key_name_admin(user)] has launched the emergency shuttle [seconds] seconds before launch.")
+							log_game("[key_name(user)] has launched the emergency shuttle in ([x], [y], [z]) [seconds] seconds before launch.")
+							GLOB.minor_announcement.Announce("The emergency shuttle will launch in 10 seconds")
+							SSshuttle.emergency.setTimer(100)
+
+				if("Repeal")
+					if(authorized.Remove(W:registered_name))
```

Comment:
`:` I know this might be copypasted, but we should remove this unsafe access `:` here, the PDA is already type checked so this shouldnt be an issue?

Reply:
Done